src/middleware/dbConnection/index.js

```diff
@@ -0,0 +1,60 @@
+import fhMbaasClient from 'fh-mbaas-client';
+import fhdb from 'fh-db';
+
+  /**
+   * @param {object} options
+   */
+export default options => {
+
+  if (!options.mbaasConf) {
+    throw new Error('mbaas configuration required');
+  }
+
+  /**
+   * Middleware to establish the database connection for a given app.
+   *
+   * @param {Object} req
+   * @param {Object} res
+   * @param {object} next
+   */
+  function middleware(req, res, next) {
+    const client = new fhMbaasClient.MbaasClient(req.envId, options.mbaasConf);
+    client.admin.apps.envVars.get({ // Get the app's environment variables by calling on fh-mbaas-client
+      domain: req.domain || {},
+      environment: req.envId,
+      appname: req.appname || {}
+    }, function(err, resp) {
+      if (err) {
+        next(err);
+      } else {
+        const appEnvVars = resp;
+        let dbparams = {};
+        if (appEnvVars && appEnvVars.FH_MONGODB_CONN_URL) {
```

Comment:
This could be rewritten as:
```
const isDedicatedDb = !!(appEnVars && appEnVars.FH_MONGODB_CONN_URL);
const dbPrams = {
	__dbperapp: isDedicatedDb,
	connectionUrl: isDedicatedDb ? appEnvVars.FH_MONGODB_CONN_URL : options.FH_MONGODB_CONN_URL
}

fhdb.createMongoCompatApi(dbparams, function(err, resp) {
    //
```

Which removes the need for the else statement